File: autoimageslider/src/main/java/com/codebyashish/autoimageslider/PageScroller.java
package com.codebyashish.autoimageslider;

import android.content.Context;
import android.view.animation.Interpolator;
import android.widget.Scroller;

public class PageScroller extends Scroller {
    int fixedDuration = 1000;
    public PageScroller(Context context) {
        super(context);
    }

    public PageScroller(Context context, Interpolator interpolator) {
        super(context, interpolator);
    }

    public PageScroller(Context context, Interpolator interpolator, boolean flywheel) {
        super(context, interpolator, flywheel);
    }

    @Override
    public void startScroll(int startX, int startY, int dx, int dy, int duration) {

        super.startScroll(startX, startY, dx, dy, fixedDuration);
    }

    @Override
    public void startScroll(int startX, int startY, int dx, int dy) {
        super.startScroll(startX, startY, dx, dy, fixedDuration);
    }

}

File: autoimageslider/src/main/java/com/codebyashish/autoimageslider/PageScroller.kt
package com.codebyashish.autoimageslider

import android.content.Context
import android.view.animation.Interpolator
import android.widget.Scroller

class PageScroller : Scroller {
    var fixedDuration = 1000

    constructor(context: Context?) : super(context)
    constructor(context: Context?, interpolator: Interpolator?) : super(context, interpolator)
    constructor(context: Context?, interpolator: Interpolator?, flywheel: Boolean) : super(
        context,
        interpolator,
        flywheel
    )

    override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int, duration: Int) {
        super.startScroll(startX, startY, dx, dy, fixedDuration)
    }

    override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int) {
        super.startScroll(startX, startY, dx, dy, fixedDuration)
    }
}
